### PR TITLE
Add redis handler

### PR DESCRIPTION
**Pull request by dankondr**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added redis protocol handler for GET requests.

Implements #33581

### CODE REVIEW

**Comment by qoega** [comment]
Intersting to see perf with Memory or Join table. And Embedded RocksDB as reference.

**Comment by alexey-milovidov** [comment]
@qoega Memory table does not support indexed lookups.
Join, Dictionary, EmbeddedRocksDB and the future AggregatingMemory are interesting.

File: tester.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import redis
```
Comment (alexey-milovidov) [inline tester.py:1]: Remove before merging.

File: src/Server/RedisHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+                LOG_DEBUG(log, "SELECT request for {} db", std::to_string(select_req.getDb()));
+
+                if (!authenticated)
+                {
+                    RedisProtocol::ErrorResponse resp(RedisProtocol::Message::NOAUTH);
+                    resp.serialize(*out);
+                    continue;
+                }
+
+                String db_name = server.config().getString(Poco::format("redis.db._%d.database", select_req.getDb()), "");
```
Comment (alexey-milovidov) [inline src/Server/RedisHandler.cpp:169]: Let's replace Poco::format with fmt::format (more efficient and easier to use).

File: src/Storages/IKVStorage.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#pragma once
```
Comment (alexey-milovidov) [inline src/Storages/IKVStorage.h:1]: Let's rename file to IKeyValueStorage.

File: src/Storages/IKVStorage.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+#include <Storages/IStorage.h>
+
+namespace DB
+{
+
+
+/// Storage that support key-value requests
```
Comment (alexey-milovidov) [inline src/Storages/IKVStorage.h:9]: Storage -> A storage.
support -> supports.

File: src/Storages/IKVStorage.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+class IKeyValueStorage : public IStorage
```
Comment (alexey-milovidov) [inline src/Storages/IKVStorage.h:10]: Another option is to add these methods to IStorage with default implementations throwing exceptions.

**Comment by qoega** [comment]
Let's revive this PR?

**Comment by vdimir** [comment]
Good PR, but on hold for now

**Pull request closed without merging**